### PR TITLE
Allow a user to provide their own Distribution configuration

### PR DESCRIPTION
**Pull request by mrclark32493**
Allow a user to provide their own Distribution configuration instead of relying on detection only.

### CODE REVIEW

**Comment by chrbayer84** [comment]
This code is needed for running MongoDB with WiredTiger on any JDK. WiredTiger needs 64 bits binaries. But BitSize is detected based on Java env variables. This change allows overrides.

File: src/main/java/de/flapdoodle/embed/process/runtime/Starter.java [hunk 1]
```diff
@@ -46,9 +46,15 @@ protected Starter(IRuntimeConfig config) {
 	}
 
 	public EXECUTABLE prepare(CONFIG config) {
-		
-		Distribution distribution = Distribution.detectFor(config.version());
-		
+		return prepare(config, null);
```
Comment (michaelmosmann) [inline src/main/java/de/flapdoodle/embed/process/runtime/Starter.java]: Can you change it to: 
  return prepare(config, Distribution.detectFor(config.version()));

File: src/main/java/de/flapdoodle/embed/process/runtime/Starter.java [hunk 2]
```diff
@@ -46,9 +46,15 @@ protected Starter(IRuntimeConfig config) {
...
+	}
+
+	public EXECUTABLE prepare(CONFIG config, Distribution distribution) {
+
+		if (distribution == null) {
```
Comment (michaelmosmann) [inline src/main/java/de/flapdoodle/embed/process/runtime/Starter.java]: You can remove this, if you change the other methods.. feel free to add a null-check:)

**Comment by mrclark32493** [comment]
Hi Michael,

I am trying to use this in a working project - can we please get a new release build?

**Comment by michaelmosmann** [comment]
Yes:)

Am 13. April 2016 18:02:18 MESZ, schrieb Matthew Clark <notifications@github.com>:
>Hi Michael,
>
>I am trying to use this in a working project - can we please get a new
>release build?  
>
>---
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/flapdoodle-oss/de.flapdoodle.embed.process/pull/37#issuecomment-209524193

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Pull request merged**